core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchQuery.java

```diff
@@ -23,19 +25,52 @@
 import org.eclipse.core.runtime.Status;
 import org.eclipse.core.runtime.SubMonitor;
 import org.eclipse.emf.ecore.EAttribute;
-import org.eclipse.emf.ecore.EClass;
 import org.eclipse.emf.ecore.EObject;
+import org.eclipse.gmf.runtime.diagram.core.util.ViewType;
+import org.eclipse.gmf.runtime.notation.Diagram;
+import org.eclipse.gmf.runtime.notation.NotationPackage;
+import org.eclipse.gmf.runtime.notation.Shape;
 import org.eclipse.jface.viewers.ITreeContentProvider;
 import org.eclipse.search.ui.ISearchQuery;
+import org.eclipse.sirius.diagram.DDiagram;
+import org.eclipse.sirius.diagram.ui.business.api.query.DDiagramGraphicalQuery;
+import org.eclipse.sirius.ext.base.Option;
+import org.eclipse.sirius.viewpoint.DRepresentation;
+import org.eclipse.sirius.viewpoint.DRepresentationDescriptor;
 import org.polarsys.capella.core.commands.preferences.util.PreferencesHelper;
 import org.polarsys.capella.core.platform.sirius.ui.navigator.viewer.CapellaNavigatorContentProvider;
+import org.polarsys.capella.core.ui.search.searchfor.item.SearchForAttributeItem;
+import org.polarsys.capella.core.ui.search.searchfor.item.SearchForClassItem;
+import org.polarsys.capella.core.ui.search.searchfor.item.SearchForNoteItem;
 
 public class CapellaSearchQuery implements ISearchQuery {
 
   private final CapellaSearchResult capellaSearchResult = new CapellaSearchResult(this);
   private final CapellaSearchSettings capellaSearchSettings;
 
-  private final ITreeContentProvider contentProvider = new CapellaNavigatorContentProvider();
+  private final ITreeContentProvider contentProvider = new CapellaNavigatorContentProvider() {
+    // Extend the search scope to include note from diagram
+    @Override
+    public Object[] getChildren(Object element) {
+      if (element instanceof DRepresentationDescriptor) {
+        DRepresentation representation = ((DRepresentationDescriptor) element).getRepresentation();

```

Comment:
This will load the diagram on T4C. So first time we use the search, all diagram will be loaded no?

Reply:
Yes, you're right. A check must be done here.